src/NumericalAlgorithms/LinearOperators/PartialDerivatives.cpp

```diff
@@ -0,0 +1,195 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+
+#include <array>
+#include <cstddef>
+#include <functional>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Transpose.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+void apply_matrix_in_first_dim(double* result, const double* const input,
```

Comment:
In theory yes, but `apply_matrices` is written so that it always does memory allocations, even if you don't need to.

Reply:
Ah :l... Well, we should probably fix that in `apply_matrices` regardless of the code duplication question. I understand if you don't want to make those changes in this PR, but if not, maybe we should make an issue so that we can go back and a) make `apply_matrices` less costly and b) deduplicate with this similar `dgemm` wrapper.

Reply:
https://github.com/sxs-collaboration/spectre/issues/2696

Reply:
Perfect :+1: